src/main/java/edu/kit/kastel/sdq/artemis4j/client/ProgrammingSubmissionDTO.java

```diff
@@ -0,0 +1,108 @@
+        @JsonProperty ParticipationDTO participation,
+        @JsonProperty String commitHash,
+        @JsonProperty boolean buildFailed,
+        @JsonProperty List<ResultDTO> results,
+        @JsonProperty ZonedDateTime submissionDate
+) {
+
+    public static List<ProgrammingSubmissionDTO> fetchAll(ArtemisClient client, long exerciseId, int correctionRound, boolean filterAssessedByTutor) throws ArtemisNetworkException {
+        var submissions = ArtemisRequest.get()
+                .path(List.of("exercises", exerciseId, "programming-submissions"))
+                .param("assessedByTutor", filterAssessedByTutor)
+                .param("correction-round", correctionRound)
+                .executeAndDecode(client, ProgrammingSubmissionDTO[].class);
+
+        List<ProgrammingSubmissionDTO> result = new ArrayList<>();
+        for (var submission : submissions) {
+            result.add(submission.fetchLongFeedback(client));
+        }
+
+        return result;
+    }
+
+    public static ProgrammingSubmissionDTO lock(ArtemisClient client, long submissionId, int correctionRound) throws ArtemisNetworkException {
+        return ArtemisRequest.get()
+                .path(List.of("programming-submissions", submissionId, "lock"))
+                .param("correction-round", correctionRound)
+                .executeAndDecode(client, ProgrammingSubmissionDTO.class)
+                .fetchLongFeedback(client);
+    }
+
+    public static Optional<ProgrammingSubmissionDTO> lockNextSubmission(ArtemisClient client, long exerciseId, int correctionRound) throws ArtemisNetworkException {
+        // Artemis returns an empty string if there is no new submission to lock
+        var submission = ArtemisRequest.get()
+                .path(List.of("exercises", exerciseId, "programming-submission-without-assessment"))
+                .param("lock", true)
+                .param("correction-round", correctionRound)
+                .executeAndDecodeMaybe(client, ProgrammingSubmissionDTO.class);
+
+        if (submission.isPresent()) {
+            submission = Optional.of(submission.get().fetchLongFeedback(client));
+        }
+
+        return submission;
+    }
+
+    public static void cancelAssessment(ArtemisClient client, long submissionId) throws ArtemisNetworkException {
+        ArtemisRequest.put()
+                .path(List.of("programming-submissions", submissionId, "cancel-assessment"))
+                .execute(client);
+    }
+
+    public static void saveAssessment(ArtemisClient client, long participationId, boolean submit, ResultDTO result) throws ArtemisNetworkException {
+        ArtemisRequest.put()
+                .path(List.of("participations", participationId, "manual-results"))
+                .param("submit", submit)
+                .body(result)
+                .execute(client);
+    }
+
+    public ProgrammingSubmissionDTO fetchLongFeedback(ArtemisClient client) throws ArtemisNetworkException {
```

Comment:
Please check what should be visible and what should be private (internal creation of the classes)